components/microsoft_onedrive/actions/upload-file/upload-file.mjs

```diff
@@ -58,7 +58,7 @@ export default {
 
     const data = fs.readFileSync(filePath);
     const extension = (await fileTypeFromBuffer(data)).ext;
-    const name = !filename.includes(".")
+    const name = !filename.includes(".") & extension
```

Comment:
Do you have sure that isn't `&&` instead of `&`?

Cc: @vunguyenhung

Reply:
@lcaresia Thanks for pointing this out, I missed an '&' operator. I have added it in the latest commit.